### PR TITLE
Enable building OpenCL-CTS with Intel DPC++/C++ compiler ICX on Windows

### PR DESCRIPTION
**Pull request by qichaogu**
Currently Intel® C++ Compiler Classic (ICC) is supported to build
OpenCL-CTS on Windows. This compiler has been discontinued since the
second half of 2023. Instead, Intel recommends that users transition to
use the LLVM-based Intel® oneAPI DPC++/C++ Compiler (ICX).
This change is to enable users to build OpenCL-CTS with ICX on Windows.

### CODE REVIEW

File: test_conformance/events/test_userevents.cpp
Comment (bashbaug) [inline test_conformance/events/test_userevents.cpp:32]: There is a single `CL_EXIT_ERROR` macro in "test_bufferreadwriterect.cpp" that doesn't require any ifdefs.  Can we use it here instead?  Even better, could we move it to a shared location and use it in both files?

File: test_common/harness/msvc9.c
Comment (bashbaug) [inline test_common/harness/msvc9.c:717]: Two small questions here:

1. Do we need any specific compiler checks for `__has_builtin`?
2. Would this be more readable with the two conditions on the same line:
`#if !defined(_WIN64) && !__has_builtin(__builtin_clz)`

File: test_conformance/events/test_userevents.cpp
Comment (qichaogu) [reply test_conformance/events/test_userevents.cpp]: Thanks for pointing this out.
Yes, it's not necessary to define multiple `CL_EXIT_ERROR` and use ifdefs here. I have removed them.
I don't move `CL_EXIT_ERROR` to a shared location. `CL_EXIT_ERROR` in this file returns -1 for a non-CL_SUCCESS code, while `CL_EXIT_ERROR` in "test_bufferreadwriterect.cpp" does not do the same.

File: test_common/harness/msvc9.c
Comment (qichaogu) [reply test_common/harness/msvc9.c:721]: 1. Yes, we need checks for `__has_builtin`. Updated.
2. I don't combine the 2 conditions together as `#if !defined(_WIN64) && !__has_builtin(__builtin_clz)` because that will require the `#else` (corresponding to `#if !defined(_WIN64)`) also need a condition `#if !__has_builtin(__builtin_clz)`.

**Review by bashbaug — COMMENTED** [review]
LGTM, ideally this should have a review from another company before merging.

**Comment by cdai2** [comment]
> LGTM, ideally this should have a review from another company before merging.

Who should be invited to review it？

**Comment by qichaogu** [comment]
> > LGTM, ideally this should have a review from another company before merging.
> 
> Who should be invited to review it？

@bashbaug Could you please invite a reviewer from another company?

**Comment by bashbaug** [comment]
@svenvh or @lakshmih would you mind taking a look at this, especially the non-CMake and non-ifdef'd parts?  Thanks!

**Comment by qichaogu** [comment]
Hi @bashbaug Could you please help merge the pull request? Thanks.

File: CMakeLists.txt
Comment (kpet) [inline CMakeLists.txt:162]: There are many shared flags here. Maybe the common flags could be added unconditionally and the others only when needed?

**Review by kpet — APPROVED** [review]
Fine with the change as is but we could deduplicate some of the flags in the CMake files.

File: CMakeLists.txt
Comment (qichaogu) [reply CMakeLists.txt:163]: Good point. Updated.

**Comment by qichaogu** [comment]
@kpet @bashbaug Could you please review again and help merge if no objection?

**Review by kpet — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as discussed in the November 5th teleconference.

**Pull request merged**
